src/main/java/Duke.java

```diff
@@ -1,10 +1,298 @@
+        if (taskCount== 0) {
+            botSpeak("No tasks has been added yet. Try adding something!");
+        } else {
+            // Prints out the list of commands with respective index number
+            printDivider();
+            System.out.println("Here are the tasks in your list:");
+            for (int i=0 ; i < taskCount; i++){
+                System.out.println((i+1) + "." + listOfTasks[i]);
+            }
+            printDivider();
+        }
+    }
+
+    /**
+     * Allows user to mark tasks as done
+     *
+     * @param command The command input by user
+     * @param listOfTasks Array containing tasks inserted by user
+     */
+    public static void doneTask(String command,Task[] listOfTasks){
+        int taskCount = Task.getNumberOfTasks();
+
+        if (isDoneValid(command)){
+            // Extract the index number of the task to be marked as done
+            int taskIndex = Integer.parseInt(command.substring(command.toLowerCase().indexOf("done") + 4).trim()) - 1;
+
+            // Make task as done if the task index inputted is at least 0 and less than the number of tasks inserted
+            if ((taskIndex >= 0) && (taskIndex < taskCount)){
+                markAsDone(listOfTasks, taskIndex);
+            } else {
+                botSpeak("Task not found. Make sure you input the correct task index number!");
+            }
+        } else {
+            botSpeak("No index number inserted. Please try again!");
+        }
+    }
+
+    /**
+     * Mark the task in the list as done
+     *
+     * @param listOfTasks Array containing tasks inserted by user
+     * @param taskIndex Index of the task indicated
+     */
+    private static void markAsDone(Task[] listOfTasks, int taskIndex) {
+        // Inform the user if the task input has already been done
+        if (listOfTasks[taskIndex].isDone){
+            botSpeak("This task has already been done! Good luck completing others!!!");
+        } else {
+            // Mark the task as done
+            listOfTasks[taskIndex].isDone = true;
+            botSpeak("Good job! I have marked this task as done:\n" +
+                    listOfTasks[taskIndex]);
+        }
+    }
+
+    /**
+     * Checks if the "done" command input by user is correct
+     * It is correct if it does not have blank space and non-digits after "done" input
+     *
+     * @param sentence String of command inserted by user
+     * @return logic true if the "done" command is valid
+     */
+    public static boolean isDoneValid(String sentence){
+        // Extract the string after "done" and convert it to array of characters
+        String stringAfterDone = sentence.substring(sentence.toLowerCase().indexOf("done") + 4).trim();
+        char[] charAfterDone = stringAfterDone.toCharArray();
+
+        // Return false if the substring after "done" only contains empty space
+        if (stringAfterDone.isEmpty()){
+            return false;
+        }
```

Comment:
Consider doing this instead "if (stringAfterDone.isEmpty()){ return false; }"

Reply:
I believe we need to create a new line even if the code is 1 line only. Check the coding standards provided.